metaflow/plugins/kfp/kfp_cli.py

```diff
@@ -242,19 +251,36 @@ def _convert_value(param: parameters.Parameter):
             fg="cyan",
         )
 
-        if wait_for_completion:
+        if argo_wait:
+            argo_path: str = shutil.which("argo")
+            run_info = flow._client.get_run(run_pipeline_result.run_id)
+            workflow_manifest = json.loads(run_info.pipeline_runtime.workflow_manifest)
+            argo_workflow_name = workflow_manifest["metadata"]["name"]
+            argo_cmd = f"{argo_path} -n {namespace} "
+            cmd = f"{argo_cmd} watch {argo_workflow_name}"
+            subprocess.run(cmd, shell=True, universal_newlines=True)
+
+            cmd = f"{argo_cmd} get {argo_workflow_name} | grep Status | awk '{{print $2}}'"
+            ret = subprocess.run(
+                cmd, shell=True, stdout=subprocess.PIPE, encoding="utf8"
+            )
+            succeeded = "Succeeded" in ret.stdout
+            show_status(run_id, kfp_run_url, obj.echo, succeeded)
```

Comment:
Just to confirm though that in a PR ~2 weeks ago I removed `argo` from the `aip-py36` images as it was adding in ~70mb of space without adding much value so might need to re-add this as I think this adds value. Ideally `argo` would just publish a smaller client (as the binary is the entire client + server currently) but seems like that'll be a longer term item!

Reply:
This would only be invoked if a CLI parameter `-aw` or `--argo-wait`. 
Actually the KF Notebook user (on Sandbox) doesn't have permissions:

```
FATA[0000] workflows.argoproj.io "helloflow-k7cxv" is forbidden: User "system:serviceaccount:aip-example-dev:default-editor" cannot get resource "workflows" in API group "argoproj.io" in the namespace "aip-example-dev" 
```

`python metaflow/tutorials/00-helloworld/helloworld.py kfp run --argo-wait`

Reply:
Great makes sense, thanks for opening that thread to get the permissions sorted for this too, let me know if you need any help getting `argo` back into the images as definitely think this is a big UX improvement! I'd even vote for setting this mode as the default once we're comfortable :)